test/csuite/random_directio/main.c

```diff
@@ -0,0 +1,1264 @@
+	th = 0;
+
+	/* Keep bitmap of "active" threads. */
+	threadmap = (0x1U << nth) - 1;
+	for (ret = 0; ret != WT_NOTFOUND && threadmap != 0;
+	     ret = cursor->next(cursor)) {
+		testutil_check(ret);
+		cursor->get_key(cursor, &gotkey);
+		gotid = (uint64_t)strtol(gotkey, &p, 10);
+		testutil_assert(*p == KEY_SEP[0]);
+		p++;
+		testutil_assert(isxdigit(*p));
+		if (isdigit(*p))
+			gotth = (uint32_t)(*p - '0');
+		else if (*p >= 'a' && *p <= 'f')
+			gotth = (uint32_t)(*p - 'a' + 10);
+		else
+			gotth = (uint32_t)(*p - 'A' + 10);
+		p++;
+		testutil_assert(*p == KEY_SEP[0]);
+		p++;
+
+		/*
+		 * See if the expected thread has finished at this point.
+		 * If so, remove it from the thread map.
+		 */
+		while (gotth != th) {
+			if ((threadmap & (0x1U << th)) != 0) {
+				threadmap &= ~(0x1U << th);
+				lastid[th] = id - 1;
+				/*
+				 * Any newly removed value in the main table
+				 * should not be present as a key in the
+				 * reverse table, since they were
+				 * transactionally inserted at the same time.
+				 */
+				gen_kv(keybuf, kvsize, id, th, large_arr[th],
+				    false);
+				check_kv(rev, keybuf, NULL, false);
+				check_schema(session, id - 1, th, flags);
+			}
+			th = (th + 1) % nth;
+			if (th == 0)
+				id++;
+		}
+		testutil_assert(gotid == id);
+		/*
+		 * Check that the key and value fully match.
+		 */
+		gen_kv(keybuf, kvsize, id, th, large_arr[th], true);
+		gen_kv(&keybuf[kvsize], kvsize, id, th, large_arr[th], false);
+		cursor->get_value(cursor, &gotvalue);
+		TEST_STREQ(keybuf, gotkey, "main table key");
+
+		/*
+		 * Every 1000th record is large.
+		 */
+		if (id % 1000 == 0)
+			TEST_STREQ(large_arr[th], gotvalue,
+			    "main table large value");
+		else
+			TEST_STREQ(&keybuf[kvsize], gotvalue,
+			    "main table value");
+
+		/*
+		 * Check the reverse file, with key/value reversed.
+		 */
+		check_kv(rev, &keybuf[kvsize], keybuf, true);
+
+		check_schema(session, id, th, flags);
+
+		/* Bump thread number and id to the next expected key. */
+		th = (th + 1) % nth;
+		if (th == 0)
+			id++;
+	}
+	printf("scanned to %" PRIu64 "\n", id);
+
+	if (LF_ISSET(SCHEMA_ALL)) {
+		/*
+		 * Check metadata to see if there are any tables
+		 * present that shouldn't be there.
+		 */
+		testutil_check(session->open_cursor(session, "metadata:", NULL,
+		    NULL, &meta));
+		while ((ret = meta->next(meta)) != WT_NOTFOUND) {
+			testutil_check(ret);
+			meta->get_key(meta, &gotkey);
+			/*
+			 * Names involved in schema testing are off the form:
+			 *   table:Axxx-t
+			 *   table:Bxxx-t
+			 * xxx corresponds to the id inserted into the main
+			 * table when the table was created, and t corresponds
+			 * to the thread id that did this.
+			 */
+			if (WT_PREFIX_SKIP(gotkey, "table:") &&
+			    (*gotkey == 'A' || *gotkey == 'B')) {
+				gotid = (uint64_t)strtol(gotkey + 1, &p, 10);
+				testutil_assert(*p == '-');
+				th = (uint32_t)strtol(p + 1, &p, 10);
+				testutil_assert(*p == '\0');
```

Comment:
Is there any checking we can or want to do on `gotid` or `th`? If not, maybe name them `dummyid, dummyth` or `unusedid, unusedth`

Reply:
Fixed, added check.